### PR TITLE
Feature/phone input

### PR DESCRIPTION
**Pull request by VilleMiekkoja**
## Description

PhoneInput component

## Related Issue

https://helsinkisolutionoffice.atlassian.net/browse/HDS-632

Closes #

https://helsinkisolutionoffice.atlassian.net/browse/HDS-632

## How Has This Been Tested?

* On developers machine locally
* Automated unit tests
* Automated visual regression tests

## Screenshots (if appropriate):

image

image

### CODE REVIEW

**Comment by ronijaakkola** [comment]
Tested React components on Storybook. They look good to go for me - waiting for accessibility audit.

**Review by laurakarhu — APPROVED** [review]
Looks good to me :) Tested in Storybook and documentation seems clear to me.

File: packages/react/src/components/phoneInput/PhoneInput.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+     */
+    useEffect(() => {
+      if (ref) {
+        if (isFunction(ref)) {
+          (ref as (instance: HTMLInputElement) => void)(inputRef.current);
+        } else {
+          // eslint-disable-next-line no-param-reassign
+          (ref as React.MutableRefObject<HTMLInputElement>).current = inputRef.current;
+        }
+      }
```
Comment (minevala) [inline packages/react/src/components/phoneInput/PhoneInput.tsx:77]: I've seen this ref-handling multiple times in our codebase. Would it be possible to make it some util-hook etc.? Or is it a too small code block to abstract?

**Review by ronijaakkola — APPROVED** [review]
Documentation looks good now!

File: packages/react/src/components/phoneInput/PhoneInput.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+import 'hds-core';
+import { InputWrapper } from '../../internal/input-wrapper/InputWrapper';
+import { TextInputProps } from '../textInput';
+import textInputStyles from '../textInput/TextInput.module.css';
+import comboseAriaDescribedBy from '../../utils/comboseAriaDescribedBy';
+
+export type PhoneInputProps = Omit<
+  TextInputProps,
+  'buttonIcon' | 'buttonAriaLabel' | 'onButtonClick' | 'children' | 'label'
```
[the remark this answers is not in the document]
Comment (VilleMiekkoja) [reply packages/react/src/components/phoneInput/PhoneInput.tsx:12]: I probably cargo-culted the omit from number input 😅 . Reasoning there was that the label of TextInput can also be a react component. And in Phone Input that requirement was not found (or atleast I did not see one in the spec), so I simplified this one and the number input one. I wonder why allowing react component as label was allowed in the first place? Tried to search it from git history but wasn't sure why it was added. Originally it used to be just string.

File: packages/react/src/components/phoneInput/PhoneInput.tsx (hunk 1 above)
Comment (VilleMiekkoja) [reply packages/react/src/components/phoneInput/PhoneInput.tsx]: - [ ] Abstract to util function

**Review by minevala — APPROVED** [review]
Good job 👍

**Pull request merged**